sdk/core/azure-core/src/main/java/com/azure/core/implementation/RestProxy.java

```diff
@@ -161,7 +161,8 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
             }
 
         } catch (Exception e) {
-            throw logger.logExceptionAsError(Exceptions.propagate(e));
+            throw logger.logExceptionAsError(new RuntimeException(
+                "Error occurs during sending request and deserializing response. Error Details: "+ e.getMessage(), e));
```

Comment:
Should use past tense for occurs, but do we want to log here as it seems it would be another generic log message after a specific log message. There might be times where we would want to ignore the logging rules.

Reply:
Try to log some content out. Have trouble to find any. Revert it back.
@anuchandy please suggest if any.